miniconf/tests/validate.rs

```diff
@@ -36,26 +37,63 @@ impl Settings {
             self.v = new;
             Ok(())
         } else {
-            Err("Must not be negative.")
+            Err("")
         }
     }
 }
 
 #[test]
 fn validate() {
     let mut s = Settings::default();
-    s.set_json("/v", "1.0".as_bytes()).unwrap();
+    s.set_json("/v", b"1.0").unwrap();
     assert_eq!(s.v, 1.0);
-    assert!(matches!(
-        s.set_json("/v", "-1.0".as_bytes()),
-        Err(Error::InvalidLeaf(1, _))
-    ));
+    assert_eq!(s.set_json("/v", b"-1.0"), Err(Error::InvalidLeaf(1, "")));
     assert_eq!(s.v, 1.0); // remains unchanged
-    s.set_json("/i/a", "1.0".as_bytes()).unwrap();
+    s.set_json("/i/a", b"1.0").unwrap();
     assert_eq!(s.i.a, 1.0);
-    assert!(matches!(
-        s.set_json("/i/a", "-1.0".as_bytes()),
-        Err(Error::InvalidInternal(1, _))
-    ));
+    assert_eq!(s.set_json("/i/a", b"-1.0"), Ok(4));
```

Comment:
Shouldn't we add a test that it now fails after this occurs because i.a is now negative?